go/vt/logutil/planetscale_logger_test.go

```diff
@@ -0,0 +1,112 @@
+package logutil
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/url"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	pslog "github.com/planetscale/log"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+
+	vtlog "vitess.io/vitess/go/vt/log"
+)
+
+// MemorySink implements zap.Sink by writing all messages to a buffer.
+// It's used to capture the logs.
+type MemorySink struct {
+	*bytes.Buffer
+}
+
+// Implement Close and Sync as no-ops to satisfy the interface. The Write
+// method is provided by the embedded buffer.
+func (s *MemorySink) Close() error { return nil }
+func (s *MemorySink) Sync() error  { return nil }
+
+func SetupLoggerWithMemSink() (sink *MemorySink, err error) {
+	// Create a sink instance, and register it with zap for the "memory"
+	// protocol.
+	sink = &MemorySink{new(bytes.Buffer)}
+	err = zap.RegisterSink("memory", func(*url.URL) (zap.Sink, error) {
+		return sink, nil
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	testLoggerConf := pslog.NewPlanetScaleConfig(pslog.DetectEncoding(), pslog.InfoLevel)
+	testLoggerConf.OutputPaths = []string{"memory://"}
+	testLoggerConf.ErrorOutputPaths = []string{"memory://"}
+	SetPlanetScaleLogger(&testLoggerConf)
+
+	return
+}
+
+func TestPSLogger_Replacing_glog(t *testing.T) {
+	type logMsg struct {
+		Level string `json:"level"`
+		Msg   string `json:"msg"`
+	}
+
+	type testCase struct {
+		name     string
+		logLevel zapcore.Level
+	}
+
+	// Given
+	dummyLogMessage := "testing log"
+	testCases := []testCase{
+		{"log info", pslog.InfoLevel},
+		{"log warn", pslog.WarnLevel},
+		{"log error", pslog.ErrorLevel},
+	}
+
+	sink, err := SetupLoggerWithMemSink()
+	if err != nil {
```

Comment:
we like using `stretchr` for tests, so you can write

```golang
require.NoError(t, err)
```

instead of the `if` statement here